modules/conflict_resolver/php/endpoints/unresolved.class.inc

```diff
@@ -0,0 +1,337 @@
+
+        $this->_cache = new \LORIS\Http\Response\JsonResponse($body);
+
+        return $this->_cache;
+    }
+
+    /**
+     * Handle POST request to fix conflicts. It will instantiate the instruments
+     * and change the filed name to the correct value, then rescore them. It will
+     * also remove the conflict_unresolved row and add a conflicts_resolved one.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface
+     */
+    private function _handlePOST(ServerRequestInterface $request): ResponseInterface
+    {
+        $user = $request->getAttribute('user');
+        $data = json_decode((string) $request->getBody(), true);
+
+        $conflictid    = $data['conflicid'] ?? null;
+        $correctanswer = $data['correctanswer'] ?? null;
+
+        if ($conflictid === null || $correctanswer === null) {
+            return new \LORIS\Http\Response\BadRequest('Missing field(s)');
+        }
+
+        $db = \NDB_Factory::singleton()->database();
+
+        // Select CommentID1 if the correct answer is the one from CommentID2
+        // (and vice versa.)
+        // This is using UNION on conflicts_resolved because a second query on the
+        // same ConflictID would not work if the conflicts_unresolved row has
+        // previouly been deleted.
+        $conflict = $db->pselectRow(
+            '
+             SELECT
+               TableName,
+               FieldName,
+               CASE :v_value
+                 WHEN Value1 THEN CommentId2
+                 ELSE CommentId1
+               END as CID
+             FROM
+               conflicts_unresolved
+             WHERE
+               ConflictID = :v_conflictid
+             UNION DISTINCT
+             SELECT
+               TableName,
+               FieldName,
+               CASE :v_value  
+                 WHEN OldValue1 THEN CommentId2
+                 ELSE CommentId1
+               END as CID
+             FROM
+               conflicts_resolved
+             WHERE
+               ConflictID = :v_conflictid
+            ',
+            array(
+             'v_conflictid' => $conflictid,
+             'v_value'      => $correctanswer,
+            )
+        );
+
+        if (empty($conflict)) {
+            return new \LORIS\Http\Response\NotFound(
+                'ConflictID not found.'
+            );
+        }
+
+        // Fix instrument data
+        $instrument = \NDB_BVL_Instrument::factory(
+            $conflict['TableName'],
+            $conflict['CID'],
+            '',
+            false
+        );
+
+        $instrument->_saveValues(array($conflict['FieldName'] => $correctanswer));
+
+        // Using an output buffer because the scode function prints to STDOUT
+        ob_start();
+        $instrument->score();
+        $error = ob_get_contents();
+        ob_end_clean();
+
+        if (!empty($error)) {
+            return new \LORIS\Http\Response\InternalServerError($error);
```

Comment:
@johnsaigle Any error message printed by the score function will be displayed in the swal. By default, it will be `Error running scoring algorithm` which is, as you said, not be really useful for a front-end user.